uint256.go

```diff
@@ -202,8 +202,8 @@ func (z *Int) Sub(x, y *Int) *Int {
 	return z
 }
 
-// umulStep computes (z, carry) = z + (x * y) + carry.
-func umulStep(z, x, y, carry uint64) (uint64, uint64) {
+// umulStep computes (lo + hi * 2^64) = z + (x * y) + carry.
+func umulStep(z, x, y, carry uint64) (lo, hi uint64) {
```

Comment:
If you are introducing `lo` and `hi` names, maybe use these in the implementation at some point to have `return lo, hi` in the end?

It may also be reasonable to swap these to `hi, lo` to be consistent with `bits.Mul64`.

Reply:
Right, speaking of which - those are actual variable declarations. I suppose they're removed during compilation, since they're never used. 
However, if we do assign them (from p, carry), there's a risk that the compiler fails to remove them.

Reply:
I checked both versions and they are identical, except the final results to stack pushes are reversed.
https://godbolt.org/z/B_dXKR

Reply:
I've swapped hi/lo to match bits.Mul64